src/rate_limiter/src/persist.rs

```diff
@@ -71,15 +71,18 @@ impl Persist<'_> for RateLimiter {
 
     fn restore(_: Self::ConstructorArgs, state: &Self::State) -> Result<Self, Self::Error> {
         let rate_limiter = RateLimiter {
-            // Safe to unwrap because TokenBucket::restore doesn't return errors.
-            ops: state
-                .ops
-                .as_ref()
-                .map(|ops| TokenBucket::restore((), ops).unwrap()),
-            bandwidth: state
-                .bandwidth
-                .as_ref()
-                .map(|bw| TokenBucket::restore((), bw).unwrap()),
+            ops: if let Some(ops) = state.ops.as_ref() {
+                let tb = TokenBucket::restore((), ops)?;
+                Some(tb)
+            } else {
+                None
+            },
+            bandwidth: if let Some(bw) = state.bandwidth.as_ref() {
+                let tb = TokenBucket::restore((), bw)?;
+                Some(tb)
```

Comment:
you could directly use `Some(TokenBucket::restore((), bw)?)`

Reply:
Sure, I missed that. Solved.